addons/point_of_sale/report/pos_invoice.py

```diff
@@ -23,5 +23,15 @@ def _get_report_values(self, docids, data=None):
             not_invoiced_posorders = PosOrder.browse(not_invoiced_orders_ids)
             not_invoiced_orders_names = [a.name for a in not_invoiced_posorders]
             raise UserError(_('No link to an invoice for %s.') % ', '.join(not_invoiced_orders_names))
+        docs = self.env['account.move'].sudo().browse(ids_to_print)
+        qr_code_urls = {}
+        for invoice in docs:
+            if invoice.display_qr_code:
+                new_code_url = invoice.generate_qr_code()
+                if new_code_url:
+                    qr_code_urls[invoice.id] = new_code_url
 
-        return {'docs': self.env['account.move'].sudo().browse(ids_to_print)}
+        return {
+            'docs': docs,
+            'qr_code_urls': qr_code_urls
```

Comment:
```suggestion
            'qr_code_urls': self.env['report.account.report_invoice'].sudo()._get_report_values(ids_to_print)['qr_code_urls']
```

Reply:
Looks like it's working fine. It was the function that I copied, Thanks for the tips!